### ISSUE TITLE
Issues with newest Windows 10 Fall Creator's Update

### ISSUE THREAD

**Issue by ghost** [MetaFight/PowerBuddy#22]
PC Powerbuddy crashes on upon executing, when you try to run it after the new Win 10 update.

**Comment by MetaFight** [comment]
Hi, thanks for the bug report.  I suspect this is the same I saw with #5.  If it is, it means they've deprecated the API I'm using to get and change power plans.

I think MS have released their updated SDK for the Fall Creator's Update.  I'll have a look see if I can fix this.

**Comment by ghost** [comment]
Good stuff. Thanks dev, and thanks for such a useful utility.

**Comment by xgdgsc** [comment]
Windows 10 Fall Creator's Update includes a sliding bar for power menu, maybe we don' t need Powerbuddy anymore.

**Comment by MetaFight** [comment]
That's a very big possibility since it looks like the old concept of "Power Plans" has been retired..

That means all Power Buddy could offer is a very similar slider.  Not really worth a 3rd party app if you ask me :)

I'll look into updating the installer to list Fall Create Update and beyond as "not supported."

**Comment by cakeseller** [comment]
@MetaFight I feel that I will still need Power Buddy after that update. I have "Gaming" power mode on my laptop which sets max CPU power to 99% — that helps to deal with CPU throttling during gaming.
Slider that was introduced in Fall Creators Update is a nice feature, but still will not help me to switch fast to my Gaming power plan.

So I will extremely appreciate if you fix Power Buddy which is very helpful application for me.

Thanks for creating it :)

**Comment by MetaFight** [comment]
@cakeseller that's a scenario I hadn't considered.  Interesting.

As far as I can tell "Power Plans" as we understood them don't exist anymore.  The API I used to fetch them is gone.  Also, the `PowerCfg` utility often used to changed them by script now behaves oddly.  It won't list all the power plans by default.

The legacy Power control panel stuff still lists them, and so does the Windows Mobility Center, but when I used that it seemed to break the new Power Slider in the Win10 battery popup.

All that to say, I'm kind of confused about how Windows handles Power Plans now.  If anyone knows please let me know.

I'll do my best to make Power Buddy work with this latest update.

**Comment by jccalhoun** [comment]
I hope you figure it out. I use power buddy on my desktop and as far as I can tell there is not way to access the new power sliders on desktop.

**Comment by ghost** [comment]
The power slider is nonsense to me honestly, especially seeing that Microsoft kept the power plans, if you dig into the legacy settings. I use the plans still, and the options they provide.

I have a ThinkPad laptop with an OLED display, and it's important to me to be able to alter plans, from display dimming settings (prolonging the display lifespan) to turning it off (watching YouTube etc).

I too hope you'll figure something out dev., your app is invaluable to me!

**Comment by Deewde** [comment]
Power Slider? What's that? I'm a desktop user and I don't see a Battery icon on my taskbar. For me, PowerBuddy is an invaluable tool. It offers the convenience notebook users enjoy to change power plans.

Give desktop users some love!

**Comment by MetaFight** [comment]
Yes, there is a power slider, though the current build of Windows won't always show it.  For me it only shows it *sometimes* when I set my power plan to Power Saver.  Anyway, that's besides the point.

I just spent some time switching to another API to read power plans from Windows.  But, just like powercfg, it won't actually list all the plans by default.

Some quick googling didn't bring up much about the removal of power plans from Windows 10.

It's discussed a bit here: https://superuser.com/questions/1260031/power-plans-disappeared-after-windows-10-fall-update-1709

but no real conclusion is reached.

I'll keep reading up on this, but it's not looking good.

**Comment by MetaFight** [comment]
Version [1.3.0.3 (pre-release)](https://github.com/PerfectlyCromulentLtd/PowerBuddy/releases/tag/1.3.0.3) is available for those who want to test the fix for this issue.

**Comment by MetaFight** [comment]
I count 13 downloads so far.  Anyone have any feedback?

On my end I'm still hitting the same limitation as `powercfg`:  Windows only seems to show two power plans after a while.

**Comment by ghost** [comment]
Been away a bit, will test the new version later today thanks.

**Comment by FSund** [comment]
Well, Windows deleted all my power plans after the update, so now only Balanced remain. But that one does show up in Power Buddy :)

**Comment by Deewde** [comment]
Installed, ran it and it seems to be working fine for me. Able to properly switch between the 3 default power plans. But this is what I don't understand: 

> Windows only seems to show two power plans after a while

What does this mean? Should I expect it to stop working properly after some time and only display two plans? Which plans? And after how much time?

**Comment by MetaFight** [comment]
@Deewde what I've observed is after switching plans a few times some of the plans stop appearing in the list.  I'm not exactly sure what the logic behind this is.

I'm thinking that, as a temporary solution, I might just cache power plans and keep displaying them in PowerBuddy even if Windows stops listing them.

Hopefully this will allow PowerBuddy to work almost exactly like before.  The tradeoff, though, is that plans that have *actually* been deleted will still be listed.  Considering plans aren't deleted very often I think it's an acceptable trade-off.

**Comment by Deewde** [comment]
It should be enough, yes. Thanks for the temp-fix

**Comment by ghost** [comment]
Finally got around to testing the update, can confirm it works sufficiently fine for me as well. The trade-off mentioned is a non-issue for me, thanks dev!

**Comment by dsl101** [comment]
1.3.0.3 has fixed it for me. I always install this reg entry after an update, to get the old battery flyout back.
```
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\ImmersiveShell]
"UseWin32BatteryFlyout"=dword:00000001
```
After doing that, click on the battery and click 'More Power Options'. I do see my original power plans available, and can switch between them there, and see it changing when I change the plan from PB. So, all looks good. I will see if they start to disappear from either list :).

**Comment by edmundo096** [comment]
Working again. I experienced the trade off, after uninstalling the old and installing the new version, I could see all my plans, on both the app and Control Panel.
I was on Power saver (changed by a script that I created to help with overheating issues from the Balanced plan since Power Saver has a useful CPU max state limit), and after I selected Balanced on the PowerBuddy popup, the Power saver one disappeared from both Control Panel and the popup.
not a big deal, just re-created the Power saver plan again.
Haven't experienced the issue again since then (half day of use).
Thanks for your support maintaining PowerBuddy!

**Comment by NaweedReza** [comment]
@MetaFight I literally created a GitHub account just to thank you. I'm one of those users who heavily relies on this app as well. My laptop has a "gaming" "movie" and more power modes which i use (I customised them so the power button does different stuff)

So yeah, thanks for this great app. The way Microsoft implemented that slider is weird because it is not always there which is weird. 

Im glad i googled it and saw that you updated it for the Creators update because i havent used it since the crash.
